Update schema to 1.12 with Font InstallerType

dkbennett opened the pull request:
* Updates schema to 1.12 and adds type "font" as a valid InstallerType and NestedInstallerType in the schema.
* Adds InstallerType Font to the manifest and associated Enum conversions.
* Updated some of the NestedInstaller validation that had assumed Portable type as the only type that could have multiple installer files (fonts can also have many fonts in a package).
* Added validation for 1.12 fonts in the manifest and associated tests for the schema version.

This PR does not add any functional support for Fonts, only the validation and schema update so the new schema can be used and is valid in the related WinGet tooling.

Tested:
* Passed the AppInstallerCLITests with [ManifestValidation] and ensured a clean result.
* Added new tests for the v 1.12 manifest updated with the new installer types in the test installer manifest to ensure Fonts are correctly passing the validation.

Closes #5686 

<!-- To check a checkbox place an "x" between the brackets. e.g: [x] -->

- [x] I have signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs).
- [x] I have updated the [Release Notes](../doc/ReleaseNotes.md).
- [x] This pull request is related to an issue.

-----

JohnMcPMS commented on src/AppInstallerCommonCore/Public/winget/ManifestValidation.h:
```diff
@@ -252,6 +253,13 @@ namespace AppInstaller::Manifest
     };
 
     static const std::unordered_set<std::filesystem::path> s_AllowedPortableFiletypes = { L".exe" };
+    static const std::unordered_set<std::filesystem::path> s_AllowedFontFiletypes = {
```
I probably didn't see this during the portable review, but it would be much better if we didn't use a static that required dynamic allocation/destruction.  I would think that a `std::array<std::wstring_view>` should be able to be read-only memory (after loader fixups).

Also, we really shouldn't be defining this in the header.

dkbennett replied:
Fixed by changing to constexpr std::array<std::wstring_view, n> and putting into the anonymous namespace in ManifestValidation.cpp, for both font and portable filetypes, for read only memory with no more allocation/deallocation at runtime.

JohnMcPMS reviewed (commented):
Would it be possible to get two more commits added:
1. Move your current 1.12 schema files out and copy the 1.11 files into the 1.12 directory, otherwise unchanged.  Commit this.
2. Move your 1.12 files back in over the 1.11 files.  Commit this.

Without at least a baseline of the 1.11 schema files, it is extremely annoying to review the changes between 1.11 and 1.12.

Another option that might set things up for success in the future would be to call the 1.12 directory `latest` and set the precedent of the pattern being:
1. copy the `latest` directory to the appropriate previous version name [so next time it would be `v1.12.0`]
2. update `latest` with the new changes

If you do this, make sure to create a commit in `latest` with 1.11 before putting the 1.12 files there.

dkbennett commented on schemas/JSON/manifests/v1.12.0/manifest.locale.1.12.0.json:
```diff
@@ -0,0 +1,271 @@
+    },
+    "ManifestVersion": {
+      "type": "string",
+      "default": "1.10.0",
```
Thanks! I fixed this when I did the v1.10.0 -> latest and then updated latest to v1.12.0, which should be in the most recent commit with a much easier to see diff.

dkbennett commented on schemas/JSON/manifests/v1.12.0/manifest.version.1.12.0.json:
```diff
@@ -0,0 +1,46 @@
+    },
+    "ManifestVersion": {
+      "type": "string",
+      "default": "1.10.0",
```
Fixed

dkbennett commented on schemas/JSON/manifests/v1.12.0/manifest.defaultLocale.1.12.0.json:
```diff
@@ -0,0 +1,280 @@
+    },
+    "ManifestVersion": {
+      "type": "string",
+      "default": "1.10.0",
```
Fixed

dkbennett commented:
> Would it be possible to get two more commits added:
> 
> 1. Move your current 1.12 schema files out and copy the 1.11 files into the 1.12 directory, otherwise unchanged.  Commit this.
> 2. Move your 1.12 files back in over the 1.11 files.  Commit this.
> 
> Without at least a baseline of the 1.11 schema files, it is extremely annoying to review the changes between 1.11 and 1.12.
> 
> Another option that might set things up for success in the future would be to call the 1.12 directory `latest` and set the precedent of the pattern being:
> 
> 1. copy the `latest` directory to the appropriate previous version name [so next time it would be `v1.12.0`]
> 2. update `latest` with the new changes
> 
> If you do this, make sure to create a commit in `latest` with 1.11 before putting the 1.12 files there.
> 
> [Note, 1.11 is actually the stand in for N-1, which I now think was 1.10?]

Fixed this using your future-proofing recommendation:
- Created 'latest' folder for schema.
- Copied 1.10 into latest, renamed version to 'latest' as well in the filename.
- Committed the above.
- Moved 1.12 into latest, renamed file version to be 'latest' for exact file diffing.
- Committed that. The clean diff for easy viewing is now here: https://github.com/microsoft/winget-cli/pull/5687/commits/e0a1a012227fbb63a84336604563bc2beb500d45
- Updated the resource for 1.12 to point to the latest.
